packages/react-core/src/components/Modal/examples/ModalWithOverflowingContent.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { Modal, ModalVariant, Button } from '@patternfly/react-core';
+
+export const ModalWithOverflowingContent: React.FunctionComponent = () => {
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
+
+  const handleModalToggle = () => {
+    setIsModalOpen(prevIsModalOpen => !prevIsModalOpen);
+  };
+
+  return (
+    <React.Fragment>
+      <Button variant="primary" onClick={handleModalToggle}>
+        Show modal
+      </Button>
+      <Modal
+        tabIndex={0}
```

Comment:
Recently, @seanforyou23 made a comment on some a11y docs about this sort of update and suggested adding an accessible label. This is because the non-negative tabindex will make the container register as a form-control on rotor menus or elements lists and without an explicit label, various browsers or screen readers may label it differently/incorrectly.
Comment: https://github.com/patternfly/patternfly-org/pull/2893#discussion_r841910449

That said, I dont think we need to update the example docs - I am adding this suggestion to the a11y docs. But maybe we should add an aria-label or aria-labelledby to elements we are intentionally adding a tabindex to?

Reply:
Makes sense. I'm on board with adding an aria-label here.

Reply:
I agree with the suggestion of adding a label, but I wonder if we would need to add a role in that scenario as well? I think technically just having an `aria-label` on a `<div>` will throw an Axe error since this approach is not well supported. For example, we actually have [this issue ](https://www.patternfly.org/v4/components/wizard/react/basic/)right now on our Wizard. I'm not sure which role would make sense in this scenario though. We could do a basic "region" role, but there might be [a better sectioning role](https://developer.mozilla.org/en-US/docs/Web/HTML/Element#content_sectioning). It might require a little bit of research to see what the best approach is to add a label.

Reply:
@jessiehuff from the research I've done so far I think the ideal role might depend on what the modal is for / what the content in the modal is.

For that reason I think the best approach here is probably to add another prop that the consumer can use to apply a more suitable role if applicable, but default to having it as a "region" if an aria-label is applied to ensure that (at the least) Axe errors aren't caused. For a little additional context from my research [Carbon](https://carbondesignsystem.com/components/modal/usage#live-demo) sets the role to "region" in their modal if it's scrollable. Thoughts?

Reply:
Also @nicolethoen the modal currently applies passed aria-labels to the overall modal box rather than letting them go to the modal box body where the content is / the tab index is applied, so another prop will need to be added for consumers to be able to apply labels to the modal box body.